### ISSUE TITLE
Functionality to decrypt gpg pillar data

### ISSUE THREAD

**Issue by seeafish** [saltstack/kitchen-salt#244]
It would appear that if you use any gpg encrypted pillar data, which I assume is fairly common, you essentially can't use kitchen-salt for testing those states.

At best, you get a warning:
``` 
[WARNING ] Could not decrypt cipher -----BEGIN PGP MESSAGE-----
       [snip]
       -----END PGP MESSAGE-----
       , received: gpg: keyblock resource `/tmp/kitchen/etc/salt/gpgkeys/secring.gpg': No such file or directory
       gpg: keyblock resource `/tmp/kitchen/etc/salt/gpgkeys/pubring.gpg': No such file or directory
       [GNUPG:] ENC_TO [snip] 1 0
       gpg: encrypted with RSA key, ID [snip]
       [GNUPG:] NO_SECKEY [snip]
       [GNUPG:] BEGIN_DECRYPTION
       [GNUPG:] DECRYPTION_FAILED
       gpg: decryption failed: No secret key
       [GNUPG:] END_DECRYPTION
```

However, say the value in that encrypted pillar is used to configure an authenticated repository, or anything else that the rest of your state depends on, your converge will simply fail.

I haven't tested this, but I think a workaround is simply dumping out your master's `/etc/salt/gpgkeys` directory onto the kitchen-salt instance. This isn't ideal as it's insecure but also it would need to be done before the converge starts.

I think the nature of gpg works makes this a bit of a challenge potentially beyond the scope of this project, but off the top of my head, some options:
1) An option in the provisioner to allow for an archive of `gpgkeys` directory to be loaded into `/etc/salt/gpgkeys`, unarchived and relevant config updated to point at that.
2) Use gpg symmetric key passed in as a provisioner option value.

What are your thoughts?

Edits: formatting.

**Comment by gtmanfred** [comment]
2) would be my preference where it exports the key from your gpg keyring where you are running the command, and drops it into /etc/salt/gpgkeys.

I will see what I can do about this, I will have some free time over the next couple days.

also, hit me up on facebook or linkedin so we can catch up.

Thanks,
Daniel

**Comment by gtmanfred** [comment]
This has been implemented by #263 and is now in `v0.4.0`

use `gpg_key` and `gpg_home` to configure this.
